### PR TITLE
[Draft] Local ZeroCost - JupyterNotebook compatibility update

### PR DESCRIPTION
**Pull request by ctr26**
Things to note:
- Requires `ipyfilechooser`, but not necessarily in notebook body
- Uses repo2docker architecture for creating consistent containers in`.binder`
- ZeroCost has path hardcoding to `/content `which doesn't work well with most "local" systems
 - Add filechooser/switching for this

TODO
- Merge newest commits to StarDist
- Documentation for: conda, conda+gpu, repo2docker, repo2docker+gpu
- Update all notebooks
- [Potentially] Add testing

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule ".binder"]
+	path = .binder
+	url = https://github.com/ctr26/basic-gpu-binder
```
Comment (ctr26) [inline .gitmodules:3]: Need to make this raw vs submodule

**Comment by ctr26** [comment]
https://github.com/HenriquesLab/ZeroCostDL4Mic/blob/jupyter/.binder/README.md

@guijacquemet 

Here are the instructions for building ZeroCost locally using conda/repo2docker. Please check and feed back. 

Thanks

**Comment by esgomezm** [comment]
- https://binder.bioimagearchive.org/
Plan: 
- Create an updated version of stardist notebook that runs in binder and in colab
- Update the wiki with the badge for those notebooks that are compatible with binder
- Link instructions in the wiki to run binder locally

**Comment by esgomezm** [comment]
Hey @ctr26 
I'm testing StarDist in binder and looks cool :) 

Still, there are some things that need to be fixed to be compatible with colab as well or to even run in binder. I updated the notebook with some of these changes but still it's not working.

- I get some errors with the environment that I'm not sure how I could fix them as it seems these should be done in the preinstallation of binder
  Screenshot 2022-10-27 at 15 25 00

- For the parameters, I guess that GPU limit is a param that you saw elsewhere? 
  Screenshot 2022-10-27 at 15 23 38

- There are many problems with the paths to colab (/content/....). Do you think we could make a relative path at the very beginning of the notebook and parameterise all? I added some of these `if COLAB==True` but it's not a very clean solution. I cannot really see a way to not go through all the notebook changing the paths though. 

Also, I cannot test very well the notebooks in binder due to the lack of GPU, but we will get to it.

**Comment by esgomezm** [comment]
Hi @ctr26 Is this PR still needed? 

@mariana-gferreira

**Comment by ctr26** [comment]
@esgomezm I get the feeling that a lot of this functionality is in DL4MicEverywhere, bits and pieces can be borrowed from this but I'd say close this PR for now and I'll keep my branch

**Pull request closed without merging**